spec/ddtrace/tasks/exec_spec.rb

```diff
@@ -54,6 +61,57 @@
         end
       end
     end
+
+    context 'when exec fails' do
+      before do
+        allow(Kernel).to receive(:exit)
```

Comment:
Seems a bit scary to stub something like this? Might not allow tests to abort if terminated... maybe okay though? I don't really have a better suggestion.

Reply:
There's two variants: `exit` and `exit!`. This doesn't touch the latter, which is the ["exit now, I don't care" version](https://rubyapi.org/3.0/o/kernel#method-i-exit-21).

The former is quite soft, it just raises an `SystemExit` exception. The docs even show how it can be caught and handled: <https://rubyapi.org/3.0/o/kernel#method-i-exit>.

So I'd say that stubbing it out is not that much of a difference from simply writing a `begin; ...; rescue Exception` which would stop it anyway, so it seems ok-ish? :)

There is a trivial alternative: just dependency-inject it. E.g.

```ruby
def initialize(args, process = Process)
```

`Process` has the same apis as `Kernel`, but that's mostly cosmetic, and then we could just pass in a mock `Process` from our tests.

Would you prefer I go for that one instead?

Reply:
I think this is probably okay as is. If it causes issue we can always go to dependency injection.